lib/rom/sql/relation/reading.rb

```diff
@@ -809,15 +809,14 @@ def coerce_conditions(conditions)
         #
         # @api private
         def __join__(type, other, join_cond = EMPTY_HASH, opts = EMPTY_HASH, &block)
-          case other
-          when Symbol, Association::Name
+          if [Symbol, Association::Name].include?(other.class)
```

Comment:
Please change this to `other.is_a?(Symbol) || other.is_a?(Association::Name)` as now it allocates redundant arrays and will be slower